extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml

```diff
@@ -54,13 +54,44 @@ spec:
                   must be unique.
                 type: string
             type: object
+            x-kubernetes-preserve-unknown-fields: true
```

Comment:
Why was this added here to `credentialsSecretRef`? 👀

Reply:
I've read the PR and it is explicitly mentioned there:

> Also, previously, the credentialSecretRef didn't have this tag x-kubernetes-preserve-unknown-fields too. Added that as well.

I'm confused why this was added in the first place, but now you say it can also be removed. What now? If it is not needed (which I think it is) then let's remove it. I suggest to then also remove it from g/mcm.

Reply:
Yes , I also didn't see any benefit for adding it at the time. Sure I will remove it from g/mcm as well.

Reply:
OK, thanks!